tests/test_checkout.py

```diff
@@ -551,6 +555,67 @@ def test_view_checkout_summary_remove_voucher(
     assert not response.context["checkout"].voucher_code
 
 
+@pytest.mark.parametrize("is_anonymous_user", (True, False))
+def test_create_order_creates_expected_events(
+    request_checkout_with_item, customer_user, shipping_method, is_anonymous_user
+):
+    checkout = request_checkout_with_item
+    checkout_user = None if is_anonymous_user else customer_user
+
+    # Ensure not events are existing prior
+    assert not OrderEvent.objects.exists()
+    assert not CustomerEvent.objects.exists()
+
+    # Prepare valid checkout
+    checkout.user = checkout_user
+    checkout.billing_address = customer_user.default_billing_address
+    checkout.shipping_address = customer_user.default_shipping_address
+    checkout.shipping_method = shipping_method
+    checkout.save()
+
+    # Place checkout
+    order = create_order(
+        checkout,
+        "tracking_code",
+        discounts=None,
+        taxes=None,
+        user=customer_user if not is_anonymous_user else AnonymousUser(),
+    )
+
+    # Ensure only two events were created, and retrieve them
+    placement_event, email_sent_event = order.events.all()  # type: OrderEvent
+
+    # Ensure the correct order event was created
+    assert placement_event.type == OrderEvents.PLACED  # is the event the expected type
```

Comment:
There is no need to create such big tests and so many asserts. You can create small unit tests for event functions - like `order_created_event`, `customer_added_to_note_order_event`, etc. Then you will be sure of the output from the function events. Then you can simply mock given event function and do `assert call`